simplq/src/components/pages/CreateQueue.jsx

```diff
@@ -11,76 +11,66 @@ import { SimplQHeader } from '../common/Header';
+  const [createInProgress, setCreateInProgress] = useState(false);
 
-  handleClick = (queueName) => {
-    if (this.state.textFieldValue === '') this.setState({ invalidMsg: 'Queue name is required' });
+  const handleClick = (queueName) => {
+    if (textFieldValue === '') setTextFieldValue('Queue name is required');
     else {
-      this.setState({ createInProgress: true });
+      setCreateInProgress(true);
       QueueService.create(queueName)
         .then((response) => {
           store.dispatch(setQueueId(response.queueId));
           store.dispatch(setQueueName(response.queueName));
           store.dispatch(setCreationStep(1));
-          this.props.history.push('/admin');
+          history.push('/admin');
         })
         .catch((err) => {
           handleApiErrors(err);
         });
-      this.setState({ createInProgress: false });
+      setCreateInProgress(false);
     }
   };
 
-  handleTextFieldChange = (e) => {
+  const handleTextFieldChange = (e) => {
     const qname = e.target.value;
-    if (isQueueNameValid(qname))
-      this.setState({
-        textFieldValue: qname,
-        invalidMsg: '',
-      });
-    else {
-      this.setState({
-        invalidMsg: "Only alphabets, numbers and '-' allowed",
-      });
+    if (isQueueNameValid(qname)) {
+      setTextFieldValue(qname);
+      setInvalidMsg('');
+    } else {
+      setInvalidMsg("Only alphabets, numbers and '-' allowed");
     }
   };
 
-  render() {
-    store.dispatch(setCreationStep(0));
-    return (
-      <div>
-        <SimplQHeader />
-        <CreatorStepper />
-        <InputField
-          placeholder="Enter a name for your new queue"
-          value={this.state.textFieldValue}
-          onChange={this.handleTextFieldChange}
-          onKeyPress={
-            (e) => handleEnterPress(e, () => this.handleClick(this.state.textFieldValue))
-            // eslint-disable-next-line react/jsx-curly-newline
-          }
-          error={this.state.invalidMsg.length > 0}
-          helperText={this.state.invalidMsg.length > 0 ? this.state.invalidMsg : ''}
-          className={styles.input}
-        />
-        <div className={styles['create-button']}>
-          {this.state.createInProgress ? (
-            <CircularProgress size={30} />
-          ) : (
-            <CreateQButton onClick={() => this.handleClick(this.state.textFieldValue)} />
-          )}
-        </div>
+  store.dispatch(setCreationStep(0)); // was inside render earlier
+
+  return (
+    <div>
+      <SimplQHeader />
+      <CreatorStepper />
+      <InputField
+        placeholder="Enter a name for your new queue"
+        value={textFieldValue}
+        onChange={handleTextFieldChange}
+        onKeyPress={
+          (e) => handleEnterPress(e, () => handleClick(textFieldValue))
+          // eslint-disable-next-line react/jsx-curly-newline
+        }
+        error={invalidMsg.length > 0}
+        helperText={invalidMsg.length > 0 ? invalidMsg : ''}
```

Comment:
Just wondering, we can just give `invalidMsg` here, will that work?

Reply:
not sure if any scenario when invalid msg is NOT a string. So should work.

Reply:
Let's give. Looks like it will be string always

Reply:
changed.